Towny/src/main/java/com/palmergames/bukkit/towny/utils/BorderUtil.java

```diff
@@ -175,4 +186,84 @@ public static boolean allowedMove(Block block, Block blockTo, @Nullable Player p
 			&& !currentTownBlock.hasResident() && !destinationTownBlock.hasResident();
 		
 	}
+
+	private static final int[][] DIRECTIONS = {{-1, 0}, {1, 0}, {0, -1}, {0, 1}};
+
+	@ApiStatus.Internal
+	public static @NotNull FloodfillResult getFloodFillableCoords(final @NotNull Town town, final @NotNull WorldCoord origin) {
+		final TownyWorld originWorld = origin.getTownyWorld();
+		if (originWorld == null || origin.hasTownBlock())
+			return FloodfillResult.fail("a");
+
+		// Filter out any coords not in the same world
+		final Set<WorldCoord> coords = new HashSet<>(town.getTownBlockMap().keySet());
+		coords.removeIf(coord -> !originWorld.equals(coord.getTownyWorld()));
+		if (coords.isEmpty())
+			return FloodfillResult.fail("No coords in the same world");
+
+		int minX = origin.getX();
+		int maxX = origin.getX();
+		int minZ = origin.getZ();
+		int maxZ = origin.getZ();
+
+		// Establish a min and max X & Z to avoid possibly looking very far
+		for (final WorldCoord coord : coords) {
+			minX = Math.min(minX, coord.getX());
+			maxX = Math.max(maxX, coord.getX());
+			minZ = Math.min(minZ, coord.getZ());
+			maxZ = Math.max(maxZ, coord.getZ());
+		}
+
+		final Set<WorldCoord> valid = new HashSet<>();
+		final Set<WorldCoord> visited = new HashSet<>();
+
+		final Queue<WorldCoord> queue = new LinkedList<>();
+		queue.offer(origin);
+		visited.add(origin);
+
+		while (!queue.isEmpty()) {
+			final WorldCoord current = queue.poll();
+
+			valid.add(current);
+
+			for (final int[] direction : DIRECTIONS) {
+				final int xOffset = direction[0];
+				final int zOffset = direction[1];
+
+				final WorldCoord candidate = current.add(xOffset, zOffset);
+
+				if (!coords.contains(candidate) && (candidate.getX() >= maxX || candidate.getX() <= minX || candidate.getZ() >= maxZ || candidate.getZ() <= minZ)) {
+					return FloodfillResult.oob();
+				}
+
+				if (!candidate.hasTownBlock() && !visited.contains(candidate) && !coords.contains(candidate)) {
```

Comment:
need to check whether we're hitting a town that isn't our own here and also fail if so